### PR TITLE
Add improvements

### PR DESCRIPTION
**Pull request by jhae-de**
* Improved code/fixed code style
* Removed unused code and files
* Added the ability to translate the filter
* Added german translations

### CODE REVIEW

File: resources/lang/de/filter.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "Select option": "Auswählen",
+  "Press enter to select": "Enter, um auszuwählen",
```
Comment (ChriRas) [inline resources/lang/de/filter.json]: Maybe a better translation:
```suggestion
  "Press enter to select": "Enter drücken, um auszuwählen",
```

File: resources/lang/de/filter.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  "Selected": "Ausgewählt",
+  "Press enter to remove": "Enter, um zu entfernen"
```
Comment (ChriRas) [inline resources/lang/de/filter.json]: Maybe a better translation:
```suggestion
  "Press enter to remove": "Enter drücken, um zu entfernen"
```

File: src/FilterServiceProvider.php [hunk 3]
```diff
@@ -13,21 +12,11 @@ class FilterServiceProvider extends ServiceProvider
...
      * @return void
      */
-    public function boot()
+    public function boot(): void
     {
-        Nova::serving(function (ServingNova $event) {
-            Nova::script('nova-multiselect-filter', __DIR__.'/../dist/js/filter.js');
-            Nova::style('nova-multiselect-filter', __DIR__.'/../dist/css/filter.css');
+        Nova::serving(static function () {
+            Nova::script('nova-multiselect-filter', __DIR__ . '/../dist/js/filter.js');
```
Comment (ChriRas) [inline src/FilterServiceProvider.php]: I think we should use constants for the path and the file.

File: src/FilterServiceProvider.php [hunk 4]
```diff
@@ -13,21 +12,11 @@ class FilterServiceProvider extends ServiceProvider
...
+            Nova::translations(__DIR__ . '/../resources/lang/de/filter.json');
```
Comment (ChriRas) [inline src/FilterServiceProvider.php]: see my comment above

**Review by ChriRas — DISMISSED** [review]

**Review by Aeroek — DISMISSED** [review]

File: resources/lang/de/filter.json (hunk 1 above)
Comment (jhae-de) [reply resources/lang/de/filter.json]: I deliberately kept it short. There is not much space.

File: resources/lang/de/filter.json (hunk 2 above)
Comment (jhae-de) [reply resources/lang/de/filter.json]: I deliberately kept it short. There is not much space.

**Review by FrankGiesecke — CHANGES_REQUESTED** [review]
See the comments from @ChriRas

**Pull request merged**